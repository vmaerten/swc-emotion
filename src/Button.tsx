import styled from "@emotion/styled";



export const Child = styled.div`
  color: red;
`

export const Parent = styled.div`
  ${Child} {
    color: green;
  }
`

export const Button = styled.button`
  font-weight: bold;
  padding: 9px 11px;
  cursor: pointer;
  border: 1px solid transparent;
  border-radius: 4px;
`
