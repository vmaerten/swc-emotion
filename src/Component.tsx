import styled from "@emotion/styled";

const OneComponent = (props: any) => {
  return (
      <div>
        just a div
      </div>
  )
}

export const Child = styled(OneComponent)`
  background: #535bf2;
  color: red;
`

export const Parent = styled.div`
  right: 34px;
  ${Child} ~ & {
    left: 36px;
  }
  div {
    top : 42px;
  }
`
