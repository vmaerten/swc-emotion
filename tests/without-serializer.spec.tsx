import React from 'react'
import renderer from 'react-test-renderer'
import { Parent } from "../src/Component";
import {test, expect} from 'vitest'


test('Button renders correctly', () => {
    expect(
        renderer.create(<div><Parent> I am the parent</Parent></div>).toJSON()
).toMatchSnapshot()
})