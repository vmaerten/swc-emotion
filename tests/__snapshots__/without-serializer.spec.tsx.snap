// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button renders correctly 1`] = `
<div>
  <div
    className="css-cfs6co e16160kb1"
  >
     I am the parent
  </div>
</div>
`;
